addons/web_editor/static/src/js/editor/odoo-editor/src/commands/commands.js

```diff
@@ -199,6 +199,10 @@ export const editorCommands = {
             }
             // Grab the content of the last child block and isolate it.
             if (isBlock(container.lastChild) && !['TABLE', 'UL', 'OL'].includes(container.lastChild.nodeName)) {
+                // Empty block must contain a text node.
```

Comment:
Why? I might have understood a `<br>` but an empty text node is very fishy.

Reply:
I thought of `<br>` but it would add an `<br>` to the next node.
for ex:
contentBefore: `<p>abc[]</p><p>def</p>`
pasteHtml: `<p>ghi</p><p><p>`
contentAfter: `<p>abcghi</p><p><br>[]def</p>`

Reply:
Then that's the problem imo. It shouldn't insert the br if the block is not empty, and it should always add one if it is.